File: tests/NewCompiler/J2C_NoReturnResults/jt5.c


jasync localyou(int c, char *s) {
    while(1) {
        jsys.sleep(10000000);
        printf("############-->>> Hello YOU  %d, %s\n", c, s);
        testfunc(-10);
    }
}

void anotherfunc() {
    printf("This is a local function \n");
}

jasync testfunc(int x) {
    printf("------------ Value of x %d\n", x);
}


int main(int argc, char *argv[]) {
    anotherfunc();
    localyou(10, "local-message...");
    return 0;
}

File: tests/J2J_NoReturnResults/jt5.js
// @ToasterConfig
// Fogs: 1
// Devices: 1

let count = 10;

jasync xyz(hello: char*) {
    console.log("Message from controller ...", hello);
}

async function sleep(n) {
    return new Promise((resolve)=> {
        setTimeout(()=> {
            resolve();
        }, n);
    });
}

if (jsys.type === 'fog') {
	while (1) {
	    xyz("hi").catch((e) => console.log(e, " at ", jsys.type));
	    await sleep(1000);
	}
} else {
    while (1) {
	jsys.setLoc({lat: count, long: 2 * count});
	count = count + 5;
	await sleep(1000);
    }
}
